### DISCUSSION TITLE
Passing args in events doesn't work

### DISCUSSION THREAD

**Discussion by PawelRoman** [zauberzeug/nicegui#2129] [Q&A]
### Description

```py
from nicegui import app, ui

@ui.page('/')
def main_page():

    def on_btn_clicked(event, *args, **kwargs):
        print(event.args)
        print(args)
        print(kwargs)

    ui.button("CLICK ME").on("click", on_btn_clicked, args=['a', 'b', 'c'])

ui.run(
    title="Some app",
    port=8502,
    native=True,
    reload=True,
    window_size=(1920, 1200),
    storage_secret='my secret'
)

```

Try the code above, 
expected: args  passed to the event handler
actual: args not passed to the event handler function, neither in event.args nor directly in *args or **kwargs

**Comment by falkoschindler** [comment]
That's not how the `args` parameter is supposed to work. Have you seen https://nicegui.io/documentation/generic_events? Buttons E and F demonstrate how `args` can be used to limit the event arguments that are send by the client to the server.

**Comment by PawelRoman** [comment]
OK, so how do I pass custom value to a button click? Like for example I have ui,grid with dynamic number of rows, each row  shows data of a Person, I have columns for first name, last name, etc. and in the last column I have ui.buttons for actions: EDIT, DELETE, VIEW to edit/delete/view given person. I have person_delete, person_edit, person_view functions but I need to pass person ID on button click. There are N*3 buttons on the page where N is number of persons (which can be dynamic). How can I even do such simple use case when I can't associate a button with person ID and pass person ID to a button click event?

**Comment by falkoschindler** [comment]
There are several ways to pass additional values to an event handler:

```py
def handle_click(e: events.ClickEventArguments, i: int):
    ui.notify(f'Button {i} clicked: {e}')

for i in range(3):
    ui.button(f'Button {i}', on_click=lambda e, i=i: handle_click(e, i))

for i in range(3, 6):
    ui.button(f'Button {i}', on_click=partial(handle_click, i=i))
```
